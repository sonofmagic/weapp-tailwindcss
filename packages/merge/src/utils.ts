import type { ClassValue } from 'clsx'
import type {
  createTailwindMerge as _createTailwindMergeV3,
  extendTailwindMerge as _extendTailwindMergeV3,
  twJoin as _twJoinV3,
  twMerge as _twMergeV3,
} from 'tailwind-merge'
import type {
  createTailwindMerge as _createTailwindMergeV2,
  extendTailwindMerge as _extendTailwindMergeV2,
  twJoin as _twJoinV2,
  twMerge as _twMergeV2,
} from 'tailwind-merge-v2'
import type { CreateOptions } from './types'
import { clsx } from 'clsx'
import { escape } from 'weapp-tailwindcss/escape'

export type TailwindMergeVersion = 2 | 3
// 处理 escape 逻辑
export function noop(x: string) {
  return x
}

export interface CreateFactoryOptions<V extends TailwindMergeVersion> {
  version: V
  twMerge: V extends 3 ? typeof _twMergeV3 : typeof _twMergeV2
  twJoin: V extends 3 ? typeof _twJoinV3 : typeof _twJoinV2
  extendTailwindMerge: V extends 3 ? typeof _extendTailwindMergeV3 : typeof _extendTailwindMergeV2
  createTailwindMerge: V extends 3 ? typeof _createTailwindMergeV3 : typeof _createTailwindMergeV2
}

export function createFactory<V extends TailwindMergeVersion>(opts: CreateFactoryOptions<V>) {
  const {
    twMerge: _twMerge,
    twJoin: _twJoin,
    createTailwindMerge: _createTailwindMerge,
    extendTailwindMerge: _extendTailwindMerge,
  } = opts

  return function create(options?: CreateOptions) {
    const escapeFn = options?.escapeFn ?? escape
    const e = options?.disableEscape ? noop : escapeFn

    function twMerge(...inputs: ClassValue[]): string {
      return e(_twMerge(clsx(...inputs)))
    }

    function twJoin(...inputs: ClassValue[]): string {
      return e(_twJoin(clsx(...inputs)))
    }

    // 版本安全的 extendTailwindMerge
    const extendTailwindMerge = (...args: Parameters<typeof _extendTailwindMerge>) => {
      // @ts-ignore
      const customTwMerge = _extendTailwindMerge(...args)
      return function cn(...inputs: ClassValue[]): string {
        return e(customTwMerge(clsx(...inputs)))
      }
    }

    // 版本安全的 createTailwindMerge
    const createTailwindMerge = (...args: Parameters<typeof _createTailwindMerge>) => {
      // @ts-ignore
      const customTwMerge = _createTailwindMerge(...args)
      return function cn(...inputs: ClassValue[]): string {
        return e(customTwMerge(clsx(...inputs)))
      }
    }

    return {
      twMerge,
      twJoin,
      extendTailwindMerge,
      createTailwindMerge,
    }
  }
}
