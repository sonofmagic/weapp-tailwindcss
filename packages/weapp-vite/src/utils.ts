import path from 'node:path'
import fs from 'fs-extra'
import klaw from 'klaw'
import { addExtension, createFilter } from '@rollup/pluginutils'

// https://developers.weixin.qq.com/miniprogram/dev/framework/structure.html
// js + json
export function isAppRoot(root: string) {
  return Boolean(searchAppEntry(root))
}

// wxml + js
export function isPage(wxmlPath: string) {
  return Boolean(searchPageEntry(wxmlPath))
}

export function searchAppEntry(root: string) {
  const extensions = ['js', 'ts']
  const appJson = path.resolve(root, 'app.json')
  if (fs.existsSync(appJson)) {
    for (const ext of extensions) {
      const entryPath = path.resolve(root, addExtension('app', `.${ext}`))
      if (fs.existsSync(entryPath)) {
        return entryPath
      }
    }
  }
}

function removeFileExtension(path: string) {
  // 使用正则表达式去掉路径中的文件后缀
  return path.replace(/\.[^/.]+$/, '')
}

export function searchPageEntry(wxmlPath: string) {
  if (fs.existsSync(wxmlPath)) {
    const extensions = ['js', 'ts']
    const base = removeFileExtension(wxmlPath)
    for (const ext of extensions) {
      const entryPath = addExtension(base, `.${ext}`)
      if (fs.existsSync(entryPath)) {
        return entryPath
      }
    }
  }
}

export function isComponent(wxmlPath: string) {
  if (isPage(wxmlPath)) {
    const jsonPath = addExtension(removeFileExtension(wxmlPath), '.json')
    if (fs.existsSync(jsonPath)) {
      return fs.readJsonSync(jsonPath).component
    }
  }
}

export function getWxmlEntry(wxmlPath: string) {
  const pageEntry = searchPageEntry(wxmlPath)
  if (pageEntry) {
    const jsonPath = addExtension(removeFileExtension(wxmlPath), '.json')
    if (fs.existsSync(jsonPath) && fs.readJsonSync(jsonPath).component) {
      return {
        path: pageEntry,
        type: 'component',
      }
    }
    return {
      path: pageEntry,
      type: 'page',
    }
  }
}

export function getProjectConfig(root: string, options?: { ignorePrivate?: boolean }) {
  const baseJsonPath = path.resolve(root, 'project.config.json')
  const privateJsonPath = path.resolve(root, 'project.private.config.json')
  let baseJson = {}
  let privateJson = {}
  if (fs.existsSync(baseJsonPath)) {
    baseJson = fs.readJsonSync(baseJsonPath) || {}
  }
  else {
    throw new Error(`在 ${root} 目录下找不到 project.config.json`)
  }
  if (!options?.ignorePrivate) {
    if (fs.existsSync(privateJsonPath)) {
      privateJson = fs.readJsonSync(privateJsonPath, {
        throws: false,
      }) || {}
    }
  }

  return Object.assign({}, privateJson, baseJson)
}

export async function scanEntries(root: string, options?: { relative?: boolean }) {
  const appEntry = searchAppEntry(root)
  // TODO exclude 需要和 output 配套
  const filter = createFilter([], ['**/node_modules/**', '**/miniprogram_npm/**', 'dist/**'])

  function getPath(to: string) {
    if (options?.relative) {
      return path.relative(root, to)
    }
    return to
  }

  if (appEntry) {
    const pageEntries = new Set<string>()
    const componentEntries = new Set<string>()
    for await (const file of klaw(root, {
      filter,
    })) {
      if (file.stats.isFile()) {
        if (/\.wxml$/.test(file.path)) {
          const entry = getWxmlEntry(file.path)
          if (entry) {
            if (entry.type === 'component') {
              componentEntries.add(getPath(entry.path))
            }
            else if (entry.type === 'page') {
              pageEntries.add(getPath(entry.path))
            }
          }
        }
      }
    }

    return {
      app: getPath(appEntry),
      pages: [...pageEntries],
      components: [...componentEntries],
    }
  }
}
