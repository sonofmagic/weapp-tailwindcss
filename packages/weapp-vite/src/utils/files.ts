export function removeExt(file: string) {
  return file.replace(/\.[^./]+$/, '')
}
