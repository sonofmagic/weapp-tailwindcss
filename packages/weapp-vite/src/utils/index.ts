import { escapeStringRegexp } from '@weapp-core/shared'

export * from './scan'

const cssComment = {
  start: '/*#weapp-vite:css-start{',
  end: '}weapp-vite:css-end#*/',
}

export function extractPathFromCss(code: string) {
  const cssCommentStart = escapeStringRegexp(cssComment.start)
  const cssCommentEnd = escapeStringRegexp(cssComment.end)
  const reg = new RegExp(`${cssCommentStart}(.*?)${cssCommentEnd}`)
  const arr = code.match(reg)
  if (arr) {
    return {
      path: arr[1],
      start: arr.index ?? 0,
      end: (arr.index ?? 0) + arr[0].length,
    }
  }
}

export function addPathForCss(filepath: string, code: string) {
  return `${cssComment.start}${filepath}${cssComment.end}\n${code}`
}

export const supportCssLang = ['wxss', 'scss', 'less', 'sass', 'styl']

export const supportCssExtensions = supportCssLang.map(x => `.${x}`)
