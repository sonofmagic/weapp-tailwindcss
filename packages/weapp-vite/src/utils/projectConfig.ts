import path from 'pathe'
import fs from 'fs-extra'

export interface ProjectConfig {
  miniprogramRoot?: string
  srcMiniprogramRoot?: string
}

export function getProjectConfig(root: string, options?: { ignorePrivate?: boolean }): ProjectConfig {
  const baseJsonPath = path.resolve(root, 'project.config.json')
  const privateJsonPath = path.resolve(root, 'project.private.config.json')
  let baseJson = {}
  let privateJson = {}
  if (fs.existsSync(baseJsonPath)) {
    try {
      baseJson = fs.readJsonSync(baseJsonPath) || {}
    }
    catch {
      throw new Error(`解析 json 格式失败, project.config.json 为非法的 json 格式`)
    }
  }
  else {
    throw new Error(`在 ${root} 目录下找不到 project.config.json`)
  }
  if (!options?.ignorePrivate) {
    if (fs.existsSync(privateJsonPath)) {
      try {
        privateJson = fs.readJsonSync(privateJsonPath) || {}
      }
      catch {
        throw new Error(`解析 json 格式失败, project.private.config.json 为非法的 json 格式`)
      }
    }
  }

  return Object.assign({}, privateJson, baseJson)
}
