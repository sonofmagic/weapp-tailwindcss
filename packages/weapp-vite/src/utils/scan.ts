import path from 'pathe'
import fs from 'fs-extra'
import { addExtension, defu, isObject, removeExtension } from '@weapp-core/shared'
import type { AppEntry, Dep, Entry, SubPackage } from '../types'

export function parseJsonUseComponents(json: any) {
  const deps: Dep[] = []
  if (isObject(json.usingComponents)) {
    deps.push(...(
      Object.values(json.usingComponents) as string[]
    ).map<Dep>((x) => {
      return {
        path: x,
        type: 'component',
      }
    }))
  }
  return deps
}

export function searchAppEntry(options?: SearchAppEntryOptions): AppEntry | undefined {
  const { formatPath, root } = defu(options, {
    formatPath: (x: string) => x,
  })
  const extensions = ['js', 'ts']
  const appJsonPath = path.resolve(root, 'app.json')
  if (fs.existsSync(appJsonPath)) {
    for (const ext of extensions) {
      const entryPath = path.resolve(root, addExtension('app', `.${ext}`))
      if (fs.existsSync(entryPath)) {
        const appJson = fs.readJSONSync(appJsonPath, { throws: false })
        const deps: Dep[] = []
        if (appJson) {
          if (Array.isArray(appJson.pages)) {
            deps.push(...(appJson.pages as string[]).map<Dep>((x) => {
              return {
                path: x,
                type: 'page',
              }
            }))
          }
          if (isObject(appJson.usingComponents)) {
            deps.push(...parseJsonUseComponents(appJson))
          }
          if (Array.isArray(appJson.subPackages)) {
            // 独立分包中不能依赖主包和其他分包中的内容，包括 js 文件、template、wxss、自定义组件、插件等（使用 分包异步化 时 js 文件、自定义组件、插件不受此条限制）
            deps.push(...(appJson.subPackages as SubPackage[]).map<Dep>((x) => {
              return {
                type: 'subPackage',
                ...x,
              }
            }))
          }
        }
        return {
          jsonPath: formatPath(appJsonPath),
          json: appJson,
          path: formatPath(entryPath),
          deps,
          type: 'app',
        }
      }
    }
  }
}
// import { isCSSRequest } from 'is-css-request'
// https://developers.weixin.qq.com/miniprogram/dev/framework/structure.html
// js + json
export function isAppRoot(root: string) {
  return Boolean(searchAppEntry({
    root,
  }))
}

export function searchPageEntry(wxmlPath: string) {
  if (fs.existsSync(wxmlPath)) {
    const extensions = ['js', 'ts']
    const base = removeExtension(wxmlPath)
    for (const ext of extensions) {
      const entryPath = addExtension(base, `.${ext}`)
      if (fs.existsSync(entryPath)) {
        return entryPath
      }
    }
  }
}
// wxml + js
export function isPage(wxmlPath: string) {
  return Boolean(searchPageEntry(wxmlPath))
}

export interface SearchAppEntryOptions {
  root: string
  formatPath?: (p: string) => string
}

export function isComponent(wxmlPath: string) {
  if (isPage(wxmlPath)) {
    const jsonPath = addExtension(removeExtension(wxmlPath), '.json')
    if (fs.existsSync(jsonPath)) {
      const json = fs.readJsonSync(jsonPath, { throws: false })
      if (json && json.component) {
        return true
      }
    }
  }
  return false
}

export function getWxmlEntry(wxmlPath: string, formatPath: (p: string) => string): Entry | undefined {
  const pageEntry = searchPageEntry(wxmlPath)
  if (pageEntry) {
    const jsonPath = addExtension(removeExtension(wxmlPath), '.json')
    const finalPath = formatPath(pageEntry)
    if (fs.existsSync(jsonPath)) {
      const json = fs.readJsonSync(jsonPath, { throws: false })
      // 是否标识 component 为 true
      if (json && json.component) {
        return {
          deps: parseJsonUseComponents(json),
          path: finalPath,
          type: 'component',
          json,
          jsonPath: formatPath(jsonPath),
        }
      }
      else {
        return {
          deps: parseJsonUseComponents(json),
          path: finalPath,
          type: 'page',
          json,
          jsonPath: formatPath(jsonPath),
        }
      }
    }
    return {
      deps: [],
      path: finalPath,
      type: 'page',
      // json: undefined,
      // jsonPath: undefined,
    }
  }
}
