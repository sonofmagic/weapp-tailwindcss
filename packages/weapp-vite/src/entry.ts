import process from 'node:process'
import mm from 'micromatch'
import { defu } from '@weapp-core/shared'
import path from 'pathe'
import klaw from 'klaw'
import { defaultExcluded, getWxmlEntry, searchAppEntry } from './utils'
import type { Entry, InlineConfig } from './types'

export function createFilter(include: string[], exclude: string[], options?: mm.Options) {
  const opts = defu<mm.Options, mm.Options[]>(options, {
    ignore: exclude,
    // dot: true,
    // contains: true,
  })
  return function (id: unknown | string) {
    if (typeof id !== 'string') {
      return false
    }
    if (/\0/.test(id)) {
      return false
    }

    return mm.isMatch(id as string, include, opts)
  }
}

export async function getEntries(options: { root?: string, srcRoot?: string, outDir?: string, relative?: boolean, subPackage?: InlineConfig['subPackage'] }) {
  // build.outDir
  const { root = process.cwd(), outDir = 'dist', relative, srcRoot = '', subPackage } = options

  function formatPath(to: string) {
    if (relative) {
      return path.relative(root, to)
    }
    return path.normalize(to)
  }

  // 单独打包分包的场景
  if (subPackage) {
    const subPackageRoot = subPackage.root ?? ''
    const filter = createFilter(
      [path.join(subPackageRoot, '**/*')],
      [
        ...defaultExcluded,
        path.join(root, `${outDir}/**`),
      ],
      { cwd: root },
    )

    const pageEntries: Entry[] = []
    const componentEntries: Entry[] = []

    for await (
      const file of klaw(
        path.join(root, subPackageRoot),
        {
          filter,
        },
      )
    ) {
      if (file.stats.isFile()) {
        if (/\.wxml$/.test(file.path)) {
          const entry = getWxmlEntry(file.path, formatPath)
          if (entry) {
            //  && !walkPathsSet.has(file.path)
            // 防止重复，理论上不会
            // walkPathsSet.add(file.path)
            if (entry.type === 'component') {
              componentEntries.push(entry)
            }
            else if (entry.type === 'page') {
              pageEntries.push(entry)
            }
          }
        }
      }
    }
    return {
      pages: pageEntries,
      components: componentEntries,
    }
  }
  // 打包主包的场景
  else {
    const appEntry = searchAppEntry(root, {
      formatPath,
    })

    // TODO exclude 需要和 output 配套
    // const walkPathsSet = new Set<string>()
    if (appEntry) {
      const subPackageDeps = appEntry.deps.filter(x => x.type === 'subPackage')
      const filter = createFilter(
        [path.join(srcRoot, '**/*')],
        [
          ...defaultExcluded,
          path.join(root, `${outDir}/**`),
          ...subPackageDeps.map((x) => {
            return path.join(root, `${x.root}/**`)
          }),
        ],
        { cwd: root },
      )

      const pageEntries: Entry[] = []
      const componentEntries: Entry[] = []

      for await (
        const file of klaw(
          path.join(root, srcRoot),
          {
            filter,
          },
        )
      ) {
        if (file.stats.isFile()) {
          if (/\.wxml$/.test(file.path)) {
            const entry = getWxmlEntry(file.path, formatPath)
            if (entry) {
              //  && !walkPathsSet.has(file.path)
              // 防止重复，理论上不会
              // walkPathsSet.add(file.path)
              if (entry.type === 'component') {
                componentEntries.push(entry)
              }
              else if (entry.type === 'page') {
                pageEntries.push(entry)
              }
            }
          }
        }
      }

      return {
        app: appEntry,
        pages: pageEntries,
        components: componentEntries,
        subPackages: subPackageDeps,
        // walkPathsSet,
      }
    }
  }
}
