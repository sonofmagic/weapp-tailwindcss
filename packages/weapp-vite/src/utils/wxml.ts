import { Parser } from 'htmlparser2'

const srcImportTagsMap: Record<string, string[]> = {
  audio: ['src', 'poster'],
  video: ['src', 'poster'],
  image: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxs/01wxs-module.html
  wxs: ['src'],
  // https://developers.weixin.qq.com/miniprogram/dev/reference/wxml/import.html
  import: ['src'],
  include: ['src'],
}

export interface WxmlDep {
  tagName: string
  start: number
  end: number
  quote: string | null | undefined
  name: string
  value: string
}

export function getDeps(wxml: string) {
  const deps: WxmlDep[] = []
  let currentTagName = ''
  let importAttrs: undefined | string[]
  const parser = new Parser({
    onopentagname(name) {
      currentTagName = name
      importAttrs = srcImportTagsMap[currentTagName]
    },
    onattribute(name, value, quote) {
      if (importAttrs) {
        for (const attrName of importAttrs) {
          if (attrName === name) {
            deps.push({
              name,
              value,
              quote,
              tagName: currentTagName,
              start: parser.startIndex,
              end: parser.endIndex,
            })
            return
          }
        }
      }
    },
    // ontext(data) {
    //   console.log('ontext', data)
    // },
    onclosetag() {
      currentTagName = ''
      importAttrs = undefined
    },
  })
  parser.write(
    wxml,
  )
  parser.end()
  return {
    deps,
  }
}
