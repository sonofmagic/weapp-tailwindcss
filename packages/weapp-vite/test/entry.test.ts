import { absDirs } from './utils'
import { getEntries } from '@/entry'

describe('entry', () => {
  // function normalizeScanEntries(result: Awaited<ReturnType<typeof scanEntries>>) {
  //   return {
  //     app: path.normalize(result?.app ?? ''),
  //     pages: [...result?.pages ?? []].map(x => path.normalize(x)),
  //     components: [...result?.components ?? []].map(x => path.normalize(x)),
  //     // css: result?.css.map(x => path.normalize(x)),
  //   }
  // }
  describe('scanEntries', () => {
    it.each(absDirs)('$name', async ({ path: root }) => {
      // expect(normalizeScanEntries()).toMatchSnapshot()
      const entries = await getEntries({
        root,
        relative: true,
      })
      expect(entries).toMatchSnapshot()
    })
  })
})
