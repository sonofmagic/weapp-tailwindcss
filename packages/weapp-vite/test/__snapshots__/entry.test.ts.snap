// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`entry > scanEntries > 'native' 1`] = `
{
  "app": {
    "deps": [
      {
        "path": "pages/index/index",
        "type": "page",
      },
      {
        "path": "pages/list/index",
        "type": "page",
      },
      {
        "path": "pages/cart/index",
        "type": "page",
      },
      {
        "path": "pages/user/index",
        "type": "page",
      },
      {
        "path": "pages/details/index",
        "type": "page",
      },
      {
        "path": "pages/bindPhone/index",
        "type": "page",
      },
      {
        "path": "pages/coupon/index",
        "type": "page",
      },
      {
        "path": "pages/order/index",
        "type": "page",
      },
      {
        "path": "pages/addressList/index",
        "type": "page",
      },
      {
        "path": "pages/addressAdd/index",
        "type": "page",
      },
      {
        "path": "pages/orderDetails/index",
        "type": "page",
      },
    ],
    "path": "app.js",
    "type": "app",
  },
  "components": [],
  "pages": [
    {
      "deps": [],
      "path": "pages/addressAdd/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/addressList/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/bindPhone/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/cart/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/coupon/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/details/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/index/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/list/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/order/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/orderDetails/index.js",
      "type": "page",
    },
    {
      "deps": [],
      "path": "pages/user/index.js",
      "type": "page",
    },
  ],
  "subPackages": [],
}
`;

exports[`entry > scanEntries > 'native-skyline' 1`] = `
{
  "app": {
    "deps": [
      {
        "path": "pages/index/index",
        "type": "page",
      },
    ],
    "path": "app.js",
    "type": "app",
  },
  "components": [
    {
      "deps": [],
      "path": "components/address-book/index.js",
      "type": "component",
    },
    {
      "deps": [],
      "path": "components/navigation-bar/index.js",
      "type": "component",
    },
    {
      "deps": [],
      "path": "components/navigation-bar/navigation-bar.js",
      "type": "component",
    },
  ],
  "pages": [
    {
      "deps": [
        {
          "path": "../../components/navigation-bar",
          "type": "component",
        },
        {
          "path": "../../components/address-book",
          "type": "component",
        },
      ],
      "path": "pages/index/index.js",
      "type": "page",
    },
  ],
  "subPackages": [],
}
`;

exports[`entry > scanEntries > 'native-ts' 1`] = `undefined`;

exports[`entry > scanEntries > 'native-ts-skyline' 1`] = `undefined`;

exports[`entry > scanEntries > 'vite-native' 1`] = `
{
  "app": {
    "deps": [
      {
        "path": "pages/index/index",
        "type": "page",
      },
      {
        "path": "/components/navigation-bar/navigation-bar",
        "type": "component",
      },
      {
        "entry": "index.js",
        "name": "pack1",
        "pages": [
          "pages/cat",
          "pages/dog",
        ],
        "root": "packageA",
        "type": "subPackage",
      },
      {
        "entry": "index.ts",
        "independent": true,
        "name": "pack2",
        "pages": [
          "pages/apple",
          "pages/banana",
        ],
        "root": "packageB",
        "type": "subPackage",
      },
    ],
    "path": "app.js",
    "type": "app",
  },
  "components": [
    {
      "deps": [],
      "path": "components/navigation-bar/navigation-bar.ts",
      "type": "component",
    },
  ],
  "pages": [
    {
      "deps": [],
      "path": "pages/index/index.ts",
      "type": "page",
    },
  ],
  "subPackages": [
    {
      "entry": "index.js",
      "name": "pack1",
      "pages": [
        "pages/cat",
        "pages/dog",
      ],
      "root": "packageA",
      "type": "subPackage",
    },
    {
      "entry": "index.ts",
      "independent": true,
      "name": "pack2",
      "pages": [
        "pages/apple",
        "pages/banana",
      ],
      "root": "packageB",
      "type": "subPackage",
    },
  ],
}
`;

exports[`entry > scanEntries > 'vite-native-skyline' 1`] = `
{
  "app": {
    "deps": [
      {
        "path": "pages/index/index",
        "type": "page",
      },
    ],
    "path": "app.js",
    "type": "app",
  },
  "components": [
    {
      "deps": [],
      "path": "components/navigation-bar/navigation-bar.js",
      "type": "component",
    },
  ],
  "pages": [
    {
      "deps": [
        {
          "path": "/components/navigation-bar/navigation-bar",
          "type": "component",
        },
      ],
      "path": "pages/index/index.js",
      "type": "page",
    },
  ],
  "subPackages": [],
}
`;

exports[`entry > scanEntries > 'vite-native-ts' 1`] = `undefined`;

exports[`entry > scanEntries > 'vite-native-ts-skyline' 1`] = `undefined`;
